receivers/sns/sns.go

```diff
@@ -21,61 +23,58 @@ import (
 // alert notifications to Amazon SNS.
 type Notifier struct {
 	*receivers.Base
-	log          logging.Logger
-	tmpl         *templates.Template
-	settings     Config
-	sessionCache *awsds.SessionCache
+	log      logging.Logger
+	tmpl     *templates.Template
+	settings Config
 }
 
 func New(cfg Config, meta receivers.Metadata, template *templates.Template, logger logging.Logger) *Notifier {
 	return &Notifier{
-		Base:         receivers.NewBase(meta),
-		log:          logger,
-		tmpl:         template,
-		settings:     cfg,
-		sessionCache: awsds.NewSessionCache(),
+		Base:     receivers.NewBase(meta),
+		log:      logger,
+		tmpl:     template,
+		settings: cfg,
 	}
 }
 
 // Notify sends the alert notification to sns.
 func (s *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
-	s.log.Info("sending SNS")
-
-	awsSessionConfig := &awsds.SessionConfig{
-		Settings: s.settings.AWSAuthSettings,
-	}
-
-	session, err := s.sessionCache.GetSession(*awsSessionConfig)
+	var (
+		tmplErr error
+		data    = notify.GetTemplateData(ctx, s.tmpl, as, s.log)
+		tmpl    = notify.TmplText(s.tmpl, data, &tmplErr)
+	)
+	s.log.Info("Sending notification")
+
+	publishInput, err := s.createPublishInput(ctx, tmpl)
 	if err != nil {
 		return false, err
 	}
 
-	publishInput, err := s.createPublishInput(ctx, as...)
+	snsClient, err := s.createSNSClient(tmpl)
 	if err != nil {
-		return false, err
+		return true, err
+	}
+
+	// check template error after we use them
+	if tmplErr != nil {
+		s.log.Warn("failed to template message", "error", tmplErr.Error())
 	}
 
-	snsClient := sns.New(session, aws.NewConfig().WithEndpoint(*aws.String(s.settings.APIUrl)))
 	publishOutput, err := snsClient.Publish(publishInput)
 	if err != nil {
 		s.log.Error("Failed to publish to Amazon SNS. ", "error", err)
 	}
 
-	s.log.Debug("SNS message successfully published", "messageId", publishOutput.MessageId, "sequenceNumber", publishOutput.SequenceNumber)
+	s.log.Debug("Message successfully published", "messageId", publishOutput.MessageId, "sequenceNumber", publishOutput.SequenceNumber)
 	return true, nil
 }
 
 func (s *Notifier) SendResolved() bool {
 	return !s.GetDisableResolveMessage()
 }
 
-func (s *Notifier) createPublishInput(ctx context.Context, as ...*types.Alert) (*sns.PublishInput, error) {
-	var err error
-	tmpl, _ := templates.TmplText(ctx, s.tmpl, as, s.log, &err)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create template: %w", err)
-	}
-
+func (s *Notifier) createPublishInput(ctx context.Context, tmpl func(string) string) (*sns.PublishInput, error) {
```

Comment:
[mega-nitpick] If this function was under `createSNSClient`, the code would be easier to compare to upstream. But this is a total nit, feel free to ignore.